contracts/core/FreeBets/FreeBetsHolder.sol

```diff
@@ -69,7 +81,11 @@ contract FreeBetsHolder is Initializable, ProxyOwned, ProxyPausable, ProxyReentr
     function fund(address _user, address _collateral, uint _amount) external notPaused nonReentrant {
         require(supportedCollateral[_collateral], "Unsupported collateral");
         IERC20(_collateral).safeTransferFrom(msg.sender, address(this), _amount);
+        if (!usersWithFreeBetPerCollateral[_collateral].contains(_user)) {
```

Comment:
the code below seems reused in the batch method, so maybe extract it to common method?